src/provider/provider_file_memory.c

```diff
@@ -0,0 +1,656 @@
+#include "utils_concurrency.h"
+#include "utils_log.h"
+
+#include <umf.h>
+#include <umf/memory_provider_ops.h>
+#include <umf/providers/provider_file_memory.h>
+
+#define TLS_MSG_BUF_LEN 1024
+
+typedef struct file_last_native_error_t {
+    int32_t native_error;
+    int errno_value;
+    char msg_buff[TLS_MSG_BUF_LEN];
+} file_last_native_error_t;
+
+static __TLS file_last_native_error_t TLS_last_native_error;
+
+// helper values used only in the Native_error_str array
+#define _UMF_FILE_RESULT_SUCCESS                                               \
+    (UMF_FILE_RESULT_SUCCESS - UMF_FILE_RESULT_SUCCESS)
+#define _UMF_FILE_RESULT_ERROR_ALLOC_FAILED                                    \
+    (UMF_FILE_RESULT_ERROR_ALLOC_FAILED - UMF_FILE_RESULT_SUCCESS)
+#define _UMF_FILE_RESULT_ERROR_FREE_FAILED                                     \
+    (UMF_FILE_RESULT_ERROR_FREE_FAILED - UMF_FILE_RESULT_SUCCESS)
+#define _UMF_FILE_RESULT_ERROR_PURGE_FORCE_FAILED                              \
+    (UMF_FILE_RESULT_ERROR_PURGE_FORCE_FAILED - UMF_FILE_RESULT_SUCCESS)
+
+static const char *Native_error_str[] = {
+    [_UMF_FILE_RESULT_SUCCESS] = "success",
+    [_UMF_FILE_RESULT_ERROR_ALLOC_FAILED] = "memory allocation failed",
+    [_UMF_FILE_RESULT_ERROR_FREE_FAILED] = "memory deallocation failed",
+    [_UMF_FILE_RESULT_ERROR_PURGE_FORCE_FAILED] = "force purging failed",
+};
+
+static void file_store_last_native_error(int32_t native_error,
+                                         int errno_value) {
+    TLS_last_native_error.native_error = native_error;
+    TLS_last_native_error.errno_value = errno_value;
+}
+
+static umf_result_t
+file_translate_params(umf_file_memory_provider_params_t *in_params,
+                      file_memory_provider_t *provider) {
+    umf_result_t result;
+
+    result = os_translate_mem_protection_flags(in_params->protection,
+                                               &provider->protection);
+    if (result != UMF_RESULT_SUCCESS) {
+        LOG_ERR("incorrect memory protection flags: %u", in_params->protection);
+        return result;
+    }
+
+    result = os_translate_mem_visibility_flag(in_params->visibility,
+                                              &provider->visibility);
+    if (result != UMF_RESULT_SUCCESS) {
+        LOG_ERR("incorrect memory visibility flag: %u", in_params->visibility);
+        return result;
+    }
+
+    return UMF_RESULT_SUCCESS;
+}
+
+static umf_result_t file_get_min_page_size(void *provider, void *ptr,
+                                           size_t *page_size);
+
+static umf_result_t file_initialize(void *params, void **provider) {
+    umf_result_t ret;
+
+    if (provider == NULL || params == NULL) {
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    umf_file_memory_provider_params_t *in_params =
+        (umf_file_memory_provider_params_t *)params;
+
+    size_t page_size;
+    umf_result_t result = file_get_min_page_size(provider, NULL, &page_size);
```

Comment:
Not by accident - it works, because the `provider` argument is not used inside `file_get_min_page_size()`, so I could pass any non-NULL pointer here, but I agree that `os_get_pagesize()` is much better in this case.

Replaced with `os_get_pagesize()`.
Done.